geode-management/src/main/java/org/apache/geode/management/internal/ClientClusterManagementService.java

```diff
@@ -194,9 +197,32 @@ private void checkIsRestful(CacheElement config) {
     }
   }
 
+  private <T extends ClusterManagementResult> T assertSuccessful(T result) {
+    if (!result.isSuccessful()) {
+      throw new RemoteClusterManagementException(result);
+    }
+    return result;
+  }
+
   public boolean isConnected() {
-    return restTemplate.getForEntity(RestfulEndpoint.URI_VERSION + "/ping", String.class)
-        .getBody().equals("pong");
+    try {
+      return restTemplate.getForEntity(RestfulEndpoint.URI_VERSION + "/ping", String.class)
+          .getBody().equals("pong");
+    } catch (Exception e) {
+      return false;
+    }
+  }
+
+  /**
+   * not public because user code should *not* try to catch this exception by name, since other
+   * implementations of {@link ClusterManagementService} will throw different subtypes of
+   * RuntimeException
+   */
+  private static class RemoteClusterManagementException extends RuntimeException {
+    public RemoteClusterManagementException(ClusterManagementResult result) {
+      super(result.getStatusCode() + (StringUtils.isEmpty(result.getStatusMessage()) ? ""
```

Comment:
Perhaps instead of extending RuntimeException, we should create a checked exception type ClusterManagementException with an additional method to get the full ClusterManagementResult.  It is possible for some operations, particularly realizations across multiple members, to be "partially successful" before failing, and the caller might need access to that rich failure information to take the appropriate cleanup steps.